crates/bevy_reflect/src/serde/de.rs

```diff
@@ -194,6 +194,15 @@ impl<'a, 'de> Visitor<'de> for ReflectVisitor<'a> {
                     })?;
                     return Ok(Box::new(list));
                 }
+                type_fields::ARRAY => {
+                    let _ = type_name
```

Comment:
```suggestion
                    let _type_name = type_name
```
NIT: For consistency with the other visit cases